src/compas_view2/objects/object.py

```diff
@@ -34,3 +64,50 @@ def create(self):
 
     def edit(self):
         pass
+
+    @property
+    def translation(self):
+        return self._translation
+
+    @translation.setter
+    def translation(self, data):
+        self._translation = data
+
+    @property
+    def rotation(self):
+        return self._rotation
+
+    @rotation.setter
+    def rotation(self, data):
+        self._rotation = data
+
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, data):
+        self._scale = data
+
+    def _update_matrix(self):
+        """Update the matrix from object's translation, rotation and scale"""
+        T1 = Translation.from_vector(self.translation)
+        R1 = Rotation.from_euler_angles(self.rotation)
+        S1 = Scale.from_factors(self.scale)
+        M = T1 * R1 * S1
+        self._transformation.data = M.data
+
+    @property
+    def matrix(self):
+        """Get the updated matrix from object's translation, rotation and scale"""
+        self._update_matrix()
+        return self._transformation.matrix
+
+    @matrix.setter
+    def matrix(self, matrix):
+        """Set the object's translation, rotation and scale from given matrix, and update object's matrix"""
+        scale, _, rotation, tranlation, _ = decompose_matrix(matrix)
+        self.translation = tranlation
+        self.rotation = rotation
+        self.scale = scale
+        self._update_matrix()
```

Comment:
why first decompose and then again compose? to exclude the projection?

Reply:
The is to automatically update the translation, rotation and scale while user directly change the matrix. Decomposing and composing ensures those three vectors and matrix are always consistent. 

 If we are to assign the matrix directly then we would need to include shear and projection as object attribute too